File: app/src/main/java/com/kota/ASFramework/PageController/ASPageView.java
package com.kota.ASFramework.PageController;

import android.content.Context;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.widget.FrameLayout;

/* loaded from: classes.dex */
public class ASPageView extends FrameLayout {
  private static int _count = 0;
  private ASViewController _owner_controller;

  protected void finalize() throws Throwable {
    super.finalize();
  }

  public ASPageView(Context context) {
    super(context);
    this._owner_controller = null;
    init();
  }

  public ASPageView(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    this._owner_controller = null;
    init();
  }

  public ASPageView(Context context, AttributeSet attrs) {
    super(context, attrs);
    this._owner_controller = null;
    init();
  }

  private void init() {
  }

  public void setOwnerController(ASViewController aController) {
    this._owner_controller = aController;
  }

  public ASViewController getOwnerController() {
    return this._owner_controller;
  }

  @Override // android.view.View
  public void draw(Canvas canvas) {
    super.draw(canvas);
    int save_count = canvas.getSaveCount();
    dispatchDraw(canvas);
    canvas.restoreToCount(save_count);
  }

  @Override // android.view.View
  protected void onDraw(Canvas aCanvas) {
  }

  public void onPageAnimationStart() {
  }

  public void onPageAnimationFinished() {
  }
}

File: app/src/main/java/com/kota/asFramework/pageController/ASPageView.kt
package com.kota.asFramework.pageController

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet
import android.widget.FrameLayout

open class ASPageView : FrameLayout {
    var ownerController: ASViewController?

    constructor(context: Context) : super(context) {
        this.ownerController = null
        init()
    }

    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    ) {
        this.ownerController = null
        init()
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        this.ownerController = null
        init()
    }

    private fun init() {
    }

    // android.view.View
    override fun draw(canvas: Canvas) {
        super.draw(canvas)
        val saveCount = canvas.saveCount
        dispatchDraw(canvas)
        canvas.restoreToCount(saveCount)
    }

    // android.view.View
    override fun onDraw(canvas: Canvas) {
    }

    fun onPageAnimationStart() {
    }

    fun onPageAnimationFinished() {
    }

    companion object {
        private const val COUNT = 0
    }
}
